src/components/product/product-extractor.service.ts

```diff
@@ -44,17 +50,26 @@ export class ProductExtractor {
 }
 
 const parseProductRow =
-  (stepColumns: Record<Step, Column>) =>
+  (pnp: Pnp, stepColumns: Record<Step, Column>) =>
   (cell: Cell<PlanningSheet>, index: number): ExtractedRow => {
     const sheet = cell.sheet;
     const row = cell.row;
+    const rowIndex = row.a1 - sheet.goals.start.row.a1;
+    const progressRow = pnp.progress.goals.start.row.a1 + rowIndex;
+
     const steps = entries(stepColumns).flatMap(([step, column]) => {
       const fiscalYear = sheet.cell(column, row).asNumber;
       const fullFY = fiscalYear ? fullFiscalYear(fiscalYear) : undefined;
       // only include step if it references a fiscal year within the project
       if (!fullFY || !sheet.projectFiscalYears.intersection(fullFY)) {
         return [];
       }
```

Comment:
This logic is duplicated now in progress extraction. Would you mind abstracting to `isGoalStepPlannedInsideProject` function?

Also here we use `sheet.projectFiscalYears` but in progress we use `sheet.projectDateRange`. I believe this one is correct and we should standardize on that.